File: solution/198/solution.cpp
class Solution {
public:
    int rob(vector<int>& nums) {
        if(nums.size()==0)return 0;
        if(nums.size()==1)return nums[0];
        std::vector<int> result(nums.size(), 0);
        result[0] = nums[0];
        result[1] = std::max(nums[0], nums[1]);
        for(int i=2;i<nums.size();i++){
            result[i] = std::max(result[i-2]+nums[i], result[i-1]);
        }
        return result[nums.size()-1];
    };
};

File: solution/198/solution.py
class Solution(object):
    def rob(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        if len(nums) == 0:
            return 0
        result = []
        result.append(nums[0])
        if len(nums) > 1:
            result.append(max(nums[0], nums[1]))
        for i in xrange(2, len(nums)):
            result.append(max(result[i-1], result[i-2]+nums[i]))
        return result[len(nums)-1]

s = Solution()
print s.rob([0,0])